### DISCUSSION TITLE
Postal does not respect docker compose extra_hosts directive

### DISCUSSION THREAD

**Discussion by TheBlueKingLP** [postalserver/postal#2567] [Help with using Postal]
## Describe the bug

When I added a hostname with extra_hosts 
## To Reproduce

1. Create a docker-compose.override.yml
2. Add `extra_hosts` to each container that exists in docker-compose.yml
3. Start Postal
4. Send a test email to a domain that uses a server with the hostname listed in `extra_hosts`
5. Error due to DNS not found or wrong IP address because Postal always checks the DNS record but not the extra_hosts

## Expected behaviour

Postal will first check if the hostname exists in extra_hosts then external DNS servers

## Environment details 

 - OS: [e.g. iOS] Debian
 - Browser [e.g. chrome, safari] Not applicable
 - Version [e.g. 22] 12
 - Type [e.g. desktop, mobile etc...] Server

## Additional information/context

Add any other context about the problem here. It is particularily useful to include log extracts (after removing private information). 

Seems like the DNS resolver Postal is using does not respect /etc/hosts file in the containers added by the extra_hosts directive in the docker-compose.override.yml file.

**Comment by willpower232** [comment]
Email domains that Postal expects to receive from are configured via the Postal UI, not an extra_hosts file.

If you need to use local nameservers to resolve DNS records then there is a setting for that in the config file.

https://github.com/postalserver/postal/blob/2f62baa238fc1102706ee4acf079b7a876b05283/config/postal.defaults.yml#L17

**Reply by TheBlueKingLP** [reply]
Where in the Postal UI?
Is there any way to override a single hostname / fqdn?

**Reply by willpower232** [reply]
If you watch the video on the docs website then that should show you the basics of how it is supposed to work https://docs.postalserver.io/
